src/components/RealtedVideos/RelatedVideos.component.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useContext, useEffect } from 'react';
+import { StoreContext } from '../../utils/store/store-context';
+import VideoCard from '../VideoCard';
+import { useGetVideos } from '../../utils/hooks/useGetVideos';
+import classes from './RelatedVideos.module.css';
+
+const RelatedVideos = ({ relatedVideoId }) => {
+  const { relatedVideoList } = useContext(StoreContext);
+  const { isLoading, error, getRelatedVideos } = useGetVideos();
+
+  useEffect(() => {
+    getRelatedVideos(relatedVideoId);
+  }, []);
+
+  const printRelatedVideoList = () => {
+    return (
+      <div className={classes['related-list']}>
+        {relatedVideoList.map((video) => (
+          <VideoCard videoData={video} key={video.id} isRelatedVideo={true} />
+        ))}
+      </div>
+    );
+  };
+
+  const printRelatedVideosContent = () => {
+    return (
+      <>
+        {!isLoading && relatedVideoList.length > 0 && printRelatedVideoList()}
+        {!isLoading && relatedVideoList.length === 0 && !error && (
+          <p>Content not found</p>
+        )}
+        {!isLoading && error && <p>{error}</p>}
+        {isLoading && <p>Loading...</p>}
```

Comment:
There should be another way of handling this. A good place to start is if you have an error nothing is going to be rendered, so you can have an 
```
if( error ) return <p> { error } </p>
```
may add some styles